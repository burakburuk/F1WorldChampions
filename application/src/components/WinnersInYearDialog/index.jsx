import React from 'react';
import PropTypes from 'prop-types';
import {
    Dialog,
    DialogContent,
    DialogTitle,
} from '../Dialog';
import WinnerTableInYearConnected from '../../containers/WinnerTableInYearConnected';

const WinnersInYearDialog = (props) => {
    const { open, selectedYear, onClose } = props;
    return (
        <Dialog
            data-id="winners-dialog"
            open={!!open}>
            <DialogContent
                width={800}>
                <DialogTitle onClose={onClose}>
                    {`Champions of ${selectedYear}`}
                </DialogTitle>
                <WinnerTableInYearConnected/>
            </DialogContent>
        </Dialog>
    );
};

WinnersInYearDialog.propTypes = {
    open: PropTypes.bool,
    onClose: PropTypes.func,
    selectedYear: PropTypes.number,
};

export default WinnersInYearDialog;
