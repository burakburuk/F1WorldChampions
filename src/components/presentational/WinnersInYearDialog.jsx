import React from 'react';
import PropTypes from 'prop-types';
import {
    Dialog,
    DialogContent,
    DialogTitle,
} from './Dialog';
import WinnerTableInYearContainer from '../container/WinnerTableInYearContainer';

const WinnersInYearDialog = (props) => {
    const { open, selectedYear, onClose } = props;
    return (
        <Dialog
            open={!!open}>
            <DialogTitle onClose={onClose}>
                {`Champions of ${selectedYear}`}
            </DialogTitle>
            <DialogContent>
                <WinnerTableInYearContainer/>
            </DialogContent>
        </Dialog>
    );
};

WinnersInYearDialog.propTypes = {
    open: PropTypes.bool,
    onClose: PropTypes.func,
    selectedYear: PropTypes.number,
};

export default WinnersInYearDialog;
