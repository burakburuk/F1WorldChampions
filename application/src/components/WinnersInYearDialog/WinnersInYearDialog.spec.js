import React from 'react';
import { shallow } from 'enzyme';
import WinnersInYearDialog from '.';

it('should match with snapshot', () => {
    const wrapper = shallow(<WinnersInYearDialog />);
    expect(wrapper).toMatchSnapshot();
});
