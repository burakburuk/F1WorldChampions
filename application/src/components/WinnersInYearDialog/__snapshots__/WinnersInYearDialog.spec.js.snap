// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<Dialog
  data-id="winners-dialog"
  open={false}
>
  <DialogContent
    width={800}
  >
    <DialogTitle>
      Champions of undefined
    </DialogTitle>
    <Connect(WinnerTableInYearContainer) />
  </DialogContent>
</Dialog>
`;
